### DISCUSSION TITLE
Channels and users are null

### DISCUSSION THREAD

**Discussion by KikyoC** [discord-jda/JDA#2742] [Questions and Help]
Hello,
I'm storing in a database users id and channels id to do some stuff.

When I'm requesting to my sql server an id I'm using JDA.getTextChannelById().
The problem is that I get an error who says the channel is null.
I tried to log what return database and everything is good.

For example in this code (most strange thing) I'm trying to send an embed with a button. Note that I don't get any error or "Channel not found log". It's just not sending the embed

```java
    public void send() {
        new Thread(() -> {
            EmbedBuilder embed = new EmbedBuilder();
            embed.setTitle(name);
            embed.setDescription(description);
            embed.addField("Nombre de gagnants", String.valueOf(winnerCount), false);
            embed.addField("Fin du giveaway", "<t:" + end.getTime()/1000 + ":R>", false);
            embed.setThumbnail(AgamiBOT.getInstance().getJda().getSelfUser().getAvatarUrl());
            embed.setFooter("AGAMICLOUD");

            TextChannel channel = AgamiBOT.getInstance().getJda().getTextChannelById(this.channel);
            if(channel == null)
                logger.warning("Channel not found");
            channel.sendMessageEmbeds(embed.build()).setActionRow(new GiveawayButton(String.valueOf(id)).getButton()).queue();
        });
    }
```

But when I'm creating a new channel (and set this.channel as TextChannel) everything works fine.

**Comment by MinnDevelopment** [comment]
You're creating a thread but not starting it, or am I missing something? Why are you creating a thread for this? This just makes your code less efficient for no benefit, no?
